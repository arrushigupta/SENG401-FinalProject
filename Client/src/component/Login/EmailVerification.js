import UserContext from "../../context/UserContext";
import LoadingContext from "../../context/LoadingContext";
import { useEffect, useState, useContext } from "react";
import { useNavigate } from "react-router-dom";
import { DINOSPatch } from "../../scripts/backend-functions";
import { verifyEmailRoute } from "../../utils/Routes";
import img from "../../img/dinosM.png";

export default function EmailVerifcation({ user }) {
  const navigate = useNavigate();
  const { setLoading } = useContext(LoadingContext);
  const { setUserID } = useContext(UserContext);
  useEffect(() => {
    if (user !== "?") {
      setLoading(false);
      DINOSPatch(verifyEmailRoute + user, setLoading);

      setTimeout(() => {
        navigate("/");
      }, "1800");
    }
  }, []);

  if (user === "?") {
    return (
      <div className="flex items-center justify-center min-h-screen p-5  min-w-screen">
        <div className="">
          <img
            className="w-16 md:w-32 lg:w-48 flex justify-start mt-24 "
            src={img}
            alt="Image not loading"
          />
        </div>
        <div class="w-11  overflow-hidden inline-block">
          <div class=" h-16  rotate-45 transform origin-bottom-left" style={{ "background-color": "#2a2d35" }}></div>
        </div>
        <div className="" style={{ "background-color": "#2a2d35", "padding": "0.5rem 0.75rem 0.75rem 0.75rem", "border-radius": "25px" }}>
          <div className=" " >
            <h3 className="text-2xl text-white">
              An email has been sent to your email address.
            </h3>
            <br />
            <h3 className="text-xl text-white">
              Please check your mailbox to verify email!
            </h3>
            <br />
            <h3 className="text-xl text-white">
              Note that ucalgary email verification may take up to two hours.</h3>
            <br />
            <h3 className="text-xl text-white"> Usually the ucalgary address will quarantine this message, please click <a className="text-red-600" href="https://security.microsoft.com/quarantine">here</a> and log in to release the message.
            </h3>
          </div>
        </div>
      </div>
    );
  } else {
    return (
      <div className="flex items-center justify-center min-h-screen p-5  min-w-screen">
        <div className="">
          <img
            className="w-16 md:w-32 lg:w-48 flex justify-start mt-24 "
            src={img}
            alt="Image not loading"
          />
        </div>
       <div className="w-1/2 h-1/2" style={{ "background-color": "#2a2d35", "padding": "0.5rem 0.75rem 0.75rem 0.75rem", "border-radius": "25px" }}>
          <div className=" " >
            <h3 className="text-xl text-white">Verifying email...</h3>
          </div>
        </div>
      </div>
    );
  }
}
