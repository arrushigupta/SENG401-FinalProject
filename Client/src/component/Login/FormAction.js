export default function FormAction({
    handleSubmit,
    action='submit',
    text
}){
    return(
        <div>
            <button
                type={action}
                className="group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-red-600 hover:bg-red-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-red-500 mt-10"
                onSubmit={handleSubmit}>
                {text}
            </button>
        </div>
    )
}