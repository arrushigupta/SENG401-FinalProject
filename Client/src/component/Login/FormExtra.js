export default function FormExtra(){
    return(
        <div className="flex items-center justify-between ">
        <div className="flex items-center">
          <input
            id="remember-me"
            name="remember-me"
            type="checkbox"
            className="h-4 w-4 text-red-600 focus:ring-red-500 border-gray-300 rounded"
          />
          <label htmlFor="remember-me" className="ml-2 block text-sm text-gray-900">
            Remember me
          </label>
        </div>

        <div className="text-sm">
          <a href="#" className="font-medium text-red-600 hover:text-red-500">
            Forgot your password?
          </a>
        </div>
      </div>

    )
}