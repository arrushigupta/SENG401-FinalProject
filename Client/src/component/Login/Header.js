import {Link} from 'react-router-dom';
import img from '../../img/dinosM.png'
export default function Header({
    heading,
    paragraph,
    linkName,
    linkUrl="#"
}){
    return(
        <div className="mb-10">
            <div className="flex justify-center">
                <img className="w-1/2 h-1/2" src={img}/>
            </div>
            <h2 className="mt-2 text-center text-3xl font-extrabold text-gray-900">
                {heading}
            </h2>
            <p className="mt-2 text-center text-sm text-gray-600 mt-5">
            {paragraph} {' '}
            <Link to={linkUrl} className="font-medium text-red-600 hover:text-red-500">
                {linkName}
            </Link>
            </p>
        </div>
    )
}