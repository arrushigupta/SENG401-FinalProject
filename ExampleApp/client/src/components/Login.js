import React, { useEffect, useState, useContext } from "react"
import PropTypes from 'prop-types';
import UserContext from "../context/UserContext";
import LoadingContext from "../context/LoadingContext";
import { Button, FormControl, FormGroup, FormLabel, Input, InputLabel, MenuItem, Select } from "@mui/material"
import { AIRPost } from "../scripts/AIRBackend";
import { useNavigate } from "react-router-dom";



export default function Login() {
  const navigate = useNavigate();
  const { setLoading } = useContext(LoadingContext);
  const [username, setUserName] = useState("");
  const [password, setPassword] = useState("");
  const [invalid, setInvalid] = useState(false);
 

  const { setUserID, setUserType } = useContext(UserContext);
  // const { userID} = useContext(UserContext);

  const handleSubmit = async e => {
    AIRPost("http://localhost:3001/login", setLoading, { username, password }, "/").then(
      (response) => {
        if (response.length == 1) {
          console.log(response[0]);
          localStorage.setItem("session", JSON.stringify({ userID: response[0].Email, userType: response[0].UserType }));
          setUserID(response[0].Email);
          setUserType(response[0].UserType);
          // console.log("Cheese: " + userID);
          navigate("/")
        } else {
          setInvalid(true);
          console.log("User not found");
        }
      }
    );
  }

  return (
    <div className="flex justify-center items-center text-xl grid" >
      <h1 className="text-3xl mt-10">Please Login to Your Account</h1>
      <div className="bg-secondary w-full m-auto mt-5 p-5 rounded-xl">
        <FormGroup className="space-y-5" >
          {invalid ? <p className="text-red-500">Invalid username or password</p> : null}
          <FormControl error={username == null || username == "" || invalid}>
            <InputLabel>Username</InputLabel>
            <Input placeholder="Username" type="text" value={username} onChange={e => setUserName(e.target.value)} />
          </FormControl>

          <FormControl error={password == null || password == "" || invalid}>
            <InputLabel>Password</InputLabel>
            <Input placeholder="Password" type="password" value={password} onChange={e => setPassword(e.target.value)} />
          </FormControl>

          <Button variant="outlined" disabled={false} type="submit" onClick={handleSubmit}>Submit</Button>


        </FormGroup >
      </div>
    </div >
  )
}

