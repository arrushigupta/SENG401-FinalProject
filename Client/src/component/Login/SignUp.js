import React, { useEffect, useState, useContext } from "react";
import UserContext from "../../context/UserContext";
import LoadingContext from "../../context/LoadingContext";
import { signupFields } from "../../constants/formField";
import FormAction from "./FormAction";
import Input from "./Input";
import { DINOSPost } from "../../scripts/backend-functions";
import { useNavigate } from "react-router-dom";
import { ToastContainer, toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
import { registerRoute } from "../../utils/Routes";

const fields = signupFields;
let fieldsState = {};

fields.forEach((field) => (fieldsState[field.id] = ""));

export default function SignUp() {
  const navigate = useNavigate();
  const { setLoading } = useContext(LoadingContext);
  const { setUserID } = useContext(UserContext);

  const [signupState, setSignupState] = useState(fieldsState);

  const handleChange = (e) => {
    setSignupState({ ...signupState, [e.target.id]: e.target.value });
  };

  const handleSubmit = (e) => {
    e.preventDefault(); // this wont refresh the page

    createAccount();
  };

  const notify = (label, error) => {
    if (label === "success") {
      toast.success("User Registered", {
        position: "top-center",
        autoClose: 2000,
        hideProgressBar: false,
        closeOnClick: true,
        pauseOnHover: true,
        draggable: true,
        progress: undefined,
        theme: "light",
      });
    }
    if (label === "error") {
      toast.error(error, {
        position: "top-center",
        autoClose: 5000,
        hideProgressBar: false,
        closeOnClick: true,
        pauseOnHover: true,
        draggable: true,
        progress: undefined,
        theme: "light",
      });
    }
  };

  //handle Signup API Integration here
  const createAccount = async () => {
    if (signupState.password !== signupState.confirm_password) {
      notify("error");
      // window.alert("Password doesn't match with Confirm Password");
      setLoading(false);
      return;
    }
    try {
      DINOSPost(registerRoute, setLoading, {
        username: signupState.username,
        email: signupState.email,
        password: signupState.password,
      }).then((response) => {

        if (response.status === "success") {
          setUserID(signupState.username);
          setSignupState("");

          notify("success");
          setTimeout(() => {
            navigate("/");
          }, "3000");
        } else {
          notify("error", response.message);
        }
      });
    } catch (error) {
      console.log(error.message);
    }
  };

  return (
    <form className="mt-8 space-y-6" onSubmit={handleSubmit}>
      <div className="">
        <ToastContainer
          position="top-center"
          autoClose={5000}
          hideProgressBar={false}
          newestOnTop={false}
          closeOnClick
          rtl={false}
          pauseOnFocusLoss
          draggable
          pauseOnHover
          theme="light"
        />
        {fields.map((field) => (
          <Input
            key={field.id}
            handleChange={handleChange}
            value={signupState[field.id]}
            labelText={field.labelText}
            labelFor={field.labelFor}
            id={field.id}
            name={field.name}
            type={field.type}
            isRequired={field.isRequired}
            placeholder={field.placeholder}
          />
        ))}
        <FormAction handleSubmit={handleSubmit} text="Signup" />
      </div>
    </form>
  );
}
