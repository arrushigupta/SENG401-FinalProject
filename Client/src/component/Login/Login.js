import React, { useState, useContext } from "react";
import { loginFields } from "../../constants/formField";
import FormAction from "./FormAction";
import FormExtra from "./FormExtra";
import Input from "./Input";
import UserContext from "../../context/UserContext";
import LoadingContext from "../../context/LoadingContext";
import { DINOSPost } from "../../scripts/backend-functions";
import { useNavigate } from "react-router-dom";
import { ToastContainer, toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";

import { loginRoute } from "../../utils/Routes";

const fields = loginFields;
let fieldsState = {};
fields.forEach((field) => (fieldsState[field.id] = ""));

export default function Login() {
  const navigate = useNavigate();
  const { setLoading } = useContext(LoadingContext);
  const { setUserID } = useContext(UserContext);

  const [username, setUserName] = useState("");
  const [password, setPassword] = useState("");
  const [invalid, setInvalid] = useState(false);
  const [loginState, setLoginState] = useState(fieldsState);

  // useEffect(() => {
  //     if (localStorage.getItem("chat-app-user")) {
  //         navigate("/chat");
  //     }
  // }, []);

  const handleChange = (e) => {
    setLoginState({ ...loginState, [e.target.id]: e.target.value });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    authenticateUser();
  };

  const notify = (label) => {
    if (label === "success") {
      toast.success("User Login Succeeded", {
        position: "top-right",
        autoClose: 1000,
        hideProgressBar: false,
        closeOnClick: true,
        pauseOnHover: true,
        draggable: true,
        progress: undefined,
        theme: "light",
      });
    }
    if (label === "error") {
      toast.error("Password doesn't match", {
        position: "top-center",
        autoClose: 5000,
        hideProgressBar: false,
        closeOnClick: true,
        pauseOnHover: true,
        draggable: true,
        progress: undefined,
        theme: "light",
      });
    }
  };
  //Handle Login API Integration here
  const authenticateUser = async () => {
    try {
      DINOSPost(loginRoute, setLoading, {
        email: loginState.email,
        password: loginState.password,
      }).then((response) => {
        if (response.status === "success") {
          setUserName(loginState.username);
          setLoginState("");
          setUserID(response._id);
          localStorage.setItem("chat-app-user", JSON.stringify(response));

          notify("success");
          if (response.emailVerified) {
            setTimeout(() => {
              navigate("/home");
            }, "1800");
          } else {
            setTimeout(() => {
              navigate("/verifyEmail?user=?");
            }, "1800");
          }
        } else {
          notify("error");
        }
      });
    } catch (error) {
      notify("error");
    }
  };

  return (
    <form className="mt-8 space-y-6" onSubmit={handleSubmit}>
      <ToastContainer
        position="top-center"
        autoClose={1800}
        hideProgressBar={false}
        newestOnTop={false}
        closeOnClick
        rtl={false}
        pauseOnFocusLoss
        draggable
        pauseOnHover
        theme="light"
      />
      <div className="-space-y-px">
        {fields.map((field) => (
          <Input
            key={field.id}
            handleChange={handleChange}
            value={loginState[field.id]}
            labelText={field.labelText}
            labelFor={field.labelFor}
            id={field.id}
            name={field.name}
            type={field.type}
            isRequired={field.isRequired}
            placeholder={field.placeholder}
          />
        ))}
      </div>

      <FormExtra />
      <FormAction handleSubmit={handleSubmit} text="Login" />
    </form>
  );
}
