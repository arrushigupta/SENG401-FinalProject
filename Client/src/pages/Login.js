import Header from "../component/Login/Header"
import Login from "../component/Login/Login"

export default function LoginPage(){
    
    return(
        <>
        <div className="min-h-full h-screen-100 flex items-center justify-center py-12 px-4 sm:px-6 lg:px-8">
        <div className="max-w-md w-full space-y-8">
            <Header
                heading="Dinos Marketplace"
                paragraph="Don't have an account yet? "
                linkName="Sign Up"
                linkUrl="/signup"
                />
            <Login/>
            </div>
            </div> 
        </>
    )
}