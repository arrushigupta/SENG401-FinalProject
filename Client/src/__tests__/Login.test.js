import renderer from "react-test-renderer";
import Login from "../component/Login/Login";
import { cleanup } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
afterEach(() => {
  cleanup();
});

test("basic test to test tests", () => {
  expect(true).toBe(true);
});

test("test login page exists", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Login />
      </MemoryRouter>
    )
    .toJSON();
  console.log(tree);
  expect(tree).toMatchSnapshot();
});
