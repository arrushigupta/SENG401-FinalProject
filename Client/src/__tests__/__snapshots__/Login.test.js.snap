// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test login page exists 1`] = `
<form
  className="mt-8 space-y-6"
  onSubmit={[Function]}
>
  <div
    className="Toastify"
  />
  <div
    className="-space-y-px"
  >
    <div
      className="my-5"
    >
      <label
        className="sr-only"
        htmlFor="email"
      >
        Email address
      </label>
      <input
        className="rounded-md appearance-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 focus:outline-none focus:ring-red-500 focus:border-red-500 focus:z-10 sm:text-smundefined"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Email address"
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="my-5"
    >
      <label
        className="sr-only"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="rounded-md appearance-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 focus:outline-none focus:ring-red-500 focus:border-red-500 focus:z-10 sm:text-smundefined"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
    </div>
  </div>
  <div
    className="flex items-center justify-between "
  >
    <div
      className="flex items-center"
    >
      <input
        className="h-4 w-4 text-red-600 focus:ring-red-500 border-gray-300 rounded"
        id="remember-me"
        name="remember-me"
        type="checkbox"
      />
      <label
        className="ml-2 block text-sm text-gray-900"
        htmlFor="remember-me"
      >
        Remember me
      </label>
    </div>
    <div
      className="text-sm"
    >
      <a
        className="font-medium text-red-600 hover:text-red-500"
        href="#"
      >
        Forgot your password?
      </a>
    </div>
  </div>
  <div>
    <button
      className="group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-red-600 hover:bg-red-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-red-500 mt-10"
      onSubmit={[Function]}
      type="submit"
    >
      Login
    </button>
  </div>
</form>
`;
